src/resolvers/Mutation/assignUserTag.ts

```diff
@@ -0,0 +1,100 @@
+import { MutationResolvers } from "../../types/generatedGraphQLTypes";
+import { errors, requestContext } from "../../libraries";
+import { User, OrganizationTagUser, TagUser } from "../../models";
+import {
+  USER_NOT_FOUND_ERROR,
+  USER_NOT_AUTHORIZED_ERROR,
+  TAG_NOT_FOUND,
+  USER_DOES_NOT_BELONG_TO_TAGS_ORGANIZATION,
+  USER_ALREADY_HAS_TAG,
+} from "../../constants";
+
+export const assignUserTag: MutationResolvers["assignUserTag"] = async (
+  _parent,
+  args,
+  context
+) => {
+  const currentUser = await User.findOne({
+    _id: context.userId,
+  }).lean();
+
+  const requestUser = await User.findOne({
```

Comment:
Move this below the check for currentUser is admin/superadmin.

Reply:
Got it.